### ISSUE TITLE
Тестирование View

### ISSUE THREAD

**Issue by ZeonXX** [Arello-Mobile/Moxy#159]
Добрый день. 
Возможно ли в Moxy как-то адекватно подменить презентер на мок в тестах.
Сейчас я использую Rule который переопределяет метод get в PresenterStore и возвращает моки презентеров разбирая тег и сравнивая имена.
Это решение очевидный костыли. Но другого способа я не нашел. :)

**Comment by senneco** [comment]
Вы можете сделать метод, который будет отвечать за создание презентера и отметить его аннотаией `@ProvidePresenter`. А в тестах возвращать из этого метода другой экземпляр презентера, например мокнутый.

**Comment by ZeonXX** [comment]
Да, я понимаю. Просто не хочется в основной код засовывать методы исключительно для тестов. Отключать в тестах создание презентаторов (аннотацию InjectPresenter) и потом просто вставлять mock в поле презентатора внутри теста было бы удобно.

**Comment by xanderblinov** [comment]
отключите для тестов annotationProcessor moxy-compiler. Таким образом отключится генерация кода по `@InjectPresenter`

**Comment by senneco** [comment]
@xanderblinov так себе совет - тогда мокси не сгенерирует MoxyReflector и все вообще сломается

**Comment by xanderblinov** [comment]
@senneco а зачем тебе рефлектор, если ты не хочешь использовать презентеры вообще?

**Comment by senneco** [comment]
Эмм, вопрос про то как подставить мок-реализацию, а не то, как не использовать мокси. А если наследоваться от MvpAppCompatActivity тоо даже во время выполнения тестов пойдет обращение к MoxyReflector

**Comment by senneco** [comment]
@ZeonXX ещё есть обсуждение этой задачи в #167

**Comment by xanderblinov** [comment]
Код нужно писать так, чтобы он был тестируемый. Для этого предоставляется связка аннотаций `@InjectPresenter` + `@ProvidePresenter`

В этом случае я рекомендую использовать DI, при помощи которого будет поставляться презентер. Для тестов вы провайдите нужный презентер из компонета. Это правильный механизм тестирования. 

@senneco причем тут вообще неипользование мокси? Всяческий воркэроунд можно делать отключением процессора аннотаций и подстановкой мокнутых классов на уровне флаворов/вариантов билда
